gunpowder/roi.py

```diff
@@ -272,29 +272,32 @@ def snap_to_grid(self, voxel_size, mode='grow'):
                 Available modes are 'grow', 'shrink', and 'closest'. Defaults to
                 'grow'.
         '''
-
-        begin_in_voxel_fractions = (
-            np.asarray(self.get_begin(), dtype=np.float32)/
-            np.asarray(voxel_size))
-        end_in_voxel_fractions = (
-            np.asarray(self.get_end(), dtype=np.float32)/
-            np.asarray(voxel_size))
-
-        if mode == 'closest':
-            begin_in_voxel = np.round(begin_in_voxel_fractions)
-            end_in_voxel = np.round(end_in_voxel_fractions)
-        elif mode == 'grow':
-            begin_in_voxel = np.floor(begin_in_voxel_fractions)
-            end_in_voxel = np.ceil(end_in_voxel_fractions)
-        elif mode == 'shrink':
-            begin_in_voxel = np.ceil(begin_in_voxel_fractions)
-            end_in_voxel = np.floor(end_in_voxel_fractions)
+        if self.unbounded():
+            return self
```

Comment:
I would avoid returning self. Someone has probably made the assumption that the `Roi` returned from `snap_to_grid` is a new instance.
`unbounded()` also returns `True` if any of the axes are unbounded. In this case we probably still want to snap any unbounded axes to the appropriate voxel size, but only leave unbounded axes as they are.